lib/ansible/modules/cloud/ovirt/ovirt_role.py

```diff
@@ -0,0 +1,177 @@
+        - create_instance
+'''
+
+RETURN = '''
+ovirt_role:
+    description: "List of dictionaries describing the Roles. Role attributes are mapped to dictionary keys,
+                  all Roles attributes can be found at following url: http://ovirt.github.io/ovirt-engine-api-model/master/#types/role."
+    returned: On success.
+    type: list
+'''
+
+from ansible.module_utils.ovirt import (
+    BaseModule,
+    check_sdk,
+    convert_to_bytes,
+    create_connection,
+    equal,
+    get_dict_of_struct,
+    get_link_name,
+    get_id_by_name,
+    ovirt_full_argument_spec,
+    search_by_attributes,
+    search_by_name,
+)
+from ansible.module_utils.basic import AnsibleModule
+import traceback
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+
+class RoleModule(BaseModule):
+    def build_entity(self):
+        all_permits = self.get_all_permits()
+        return otypes.Role(
+            id=self.param('id'),
+            name=self.param('name'),
+            mutable=self.param('mutable') if self.param('mutable') else None,
+            administrative=self.param('administrative') if self.param(
+                'administrative') else None,
+            permits=[
+                otypes.Permit(id=all_permits.get(new_permit)) for new_permit in self.param('permits')
+            ]
+        )
+
+    def get_all_permits(self):
+        return dict((permit.name, permit.id) for permit in self._connection.system_service().cluster_levels_service().level_service('4.3').get().permits)
+
+    def update_check(self, entity):
+        def check_permits():
+            if self.param('permits'):
+                permits_service = self._service.service(entity.id).permits_service()
+                current = [er.name for er in permits_service.list()]
+                passed = [pr for pr in self.param('permits')]
+                if not sorted(current) == sorted(passed):
+                    # remove all
+                    for permit in permits_service.list():
+                        permits_service.permit_service(permit.id).remove()
+                    # add passed permits
+                    all_permits = self.get_all_permits()
+                    for new_permit in passed:
+                        permits_service.add(otypes.Permit(id=all_permits.get(new_permit)))
+                    return False
+            return True
+
+        return (
+            check_permits() and
+            equal(self.param('mutable'), entity.mutable) and
+            equal(self.param('administrative'), entity.administrative)
+        )
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        state=dict(
+            choices=['present', 'absent'],
+            default='present',
+        ),
+        id=dict(default=None),
+        name=dict(default=None),
+        mutable=dict(default=True, type='bool'),
+        administrative=dict(default=False, type='bool'),
+        permits=dict(type='list', default=[]),
```

Comment:
Hmm, I would use default [], because if you update only description for example of the role it would remove all pertmis I ithink

Reply:
Has this comment regarding `default` been resolved?